### PR TITLE
Skip listing Glue tables with invalid column types

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Skip listing Glue tables with invalid column types. ({issue}`16677`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java [hunk 1]
```diff
@@ -136,7 +136,17 @@ private static Column convertColumn(com.amazonaws.services.glue.model.Column glu
...
+        return new Column(glueColumn.getName(), convertType(glueColumn), Optional.ofNullable(glueColumn.getComment()));
+    }
+
+    private static HiveType convertType(com.amazonaws.services.glue.model.Column column)
+    {
+        try {
+            return HiveType.valueOf(column.getType().toLowerCase(Locale.ENGLISH));
+        }
+        catch (IllegalArgumentException e) {
+            throw new TrinoException(HIVE_INVALID_METADATA, "Glue column '%s' has invalid data type: %s".formatted(column.getName(), column.getType()));
```
Comment (hashhar) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java:148]: would column name itself be sufficient to identify the offending table?

Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java:148]: Maybe not, but it's better than the `IllegalArgumentException` you get today.

Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java]: I updated this to include the table name

**Review by hashhar — APPROVED** [review]
LGTM.

I think it's not possible to add a test since neither Trino nor Hive creates such problematic columns.

**Pull request merged**